### PR TITLE
Feature/service tree units fetch

### PR DESCRIPTION
**Pull request by mikkoisa**
- Added units fetch to service tree view. Choosing a service node checkbox will fetch the node's units
- Modified service tree service node fetching. Now service tree will load all node child nodes when a node is expanded
- Added loading screen to map during service tree units fetch

### CODE REVIEW

File: src/views/ServiceTreeView/ServiceTreeView.js [hunk 1]
```diff
@@ -111,47 +133,92 @@ const ServiceTreeView = (props) => {
...
 
+  const checkParents = (list, nodeID) => {
+    // This checks recursively if some parent node is included in given list
+    const serviceNode = services.find(service => service.id === nodeID);
+    if (!serviceNode || !serviceNode.parent) return false;
+    if (list.includes(serviceNode.parent)) {
+      return true;
+    }
+    return checkParents(list, serviceNode.parent);
+  };
```
Comment (Dremars) [inline src/views/ServiceTreeView/ServiceTreeView.js:144]: Wasn't this same code in serviceTree selector as well. Can we export that?

File: src/views/ServiceTreeView/ServiceTreeView.js [hunk 2]
```diff
@@ -111,47 +133,92 @@ const ServiceTreeView = (props) => {
...
+        setTreeOpened([...opened, node.id]);
+        if (selected.includes(node.id) || checkParents(selected, node.id)) {
+          setTreeSelected([...selected, ...data.map(item => item.id)]);
+        }
+      })
+      .catch((e) => {
+        console.warn(e);
+        dispatchNodeFetch({ type: 'end', fetching: node.id });
+      });
+  };
```
Comment (Dremars) [inline src/views/ServiceTreeView/ServiceTreeView.js:178]: I'm wondering if this kind of redux logic should be in redux folder or as a seperate  utility file specific to ServiceTreeView. This component seems to have a lot of logic within.

**Review by Dremars — CHANGES_REQUESTED** [review]
Spent some time to go through the code and it looks really good. Some minor improvements for clarity might be needed.

For accessibility I noticed that Loading texts when opening accordion element and when loading units are not read for NVDA. Also when unit loading is done there is no verbal indication that loading has completed. But the actual loading progress sound is working correctly which helps.

File: src/views/ServiceTreeView/ServiceTreeView.js (hunk 2 above)
Comment (mikkoisa) [reply src/views/ServiceTreeView/ServiceTreeView.js:178]: Yes, it seems that this fetch does not actually interact directly with the component anymore, all fetch result data is passed through redux to the component. I'll move it to redux.

File: src/views/ServiceTreeView/ServiceTreeView.js (hunk 1 above)
Comment (mikkoisa) [reply src/views/ServiceTreeView/ServiceTreeView.js:144]: Good call, will change.

**Comment by mikkoisa** [comment]
Need to figure out how to handle selecting checkbox with screenreader. Loading units to map is useless to screenreader users.

**Pull request closed without merging**